src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -1242,60 +1246,62 @@ def get_updated_endpoints(endpoints):
 
         zonegroup_ip_url = ','.join(get_updated_endpoints(zonegroup_endpoints))
         zone_ip_url = ','.join(get_updated_endpoints(zone_endpoints))
-        try:
-            # Create the realm and zonegroup
-            self.update_progress(
-                f"Creating realm: {realm_name}, zonegroup: {zonegroup_name} and zone: {zone_name}")
-            logger.info("Creating realm: %s", realm_name)
-            rgw_multisite_instance.create_realm(realm_name=realm_name, default=True)
-            logger.info("Creating zonegroup: %s", zonegroup_name)
-            rgw_multisite_instance.create_zonegroup(realm_name=realm_name,
+
+        if not selectedRealmName:
+            try:
```

Comment:
just for readability , we can move this logic to a function as now need to scroll to match next indented block (for this if)

Reply:
Thanks for the suggestion. I agree, the existing code was a bit difficult to read. I have moved the code into smaller methods for better readability